docs/yul_language/statements.rst

```diff
@@ -0,0 +1,118 @@
+Statements
+==========
+
+For-loop
+________
+
+For-loops are completely supported in Solang. ``continue`` and ``break`` statements are also supported.
+
+.. code-block:: yul
+
+    {
+        function foo()
+        {
+           for {let i := 0} lt(i, 10) {i := add(i, 10)} {
+                let p := funcCall(i, 10)
+                if eq(p, 5) {
+                    continue
+                }
+
+                if eq(p, 90) {
+                    break
+                }
+            }
+        }
+    }
+
+If-block
+________
+
+If-block conditions in Yul cannot have an `else`. They act only as a branch if the condition is true
+and are totally supported in Solang.
+
+.. code-block:: yul
+
+    {
+        if eq(5, 4) {
+            funcCall(4, 3)
+        } // There cannot be an 'else' here
+    }
+
+
+Switch
+_______
+
+Switch statements are not yet supported in Solang. If there is urgent need to support them,
+please, file a GitHub issue in the official repository.
+
+Blocks
+______
+
+There can be blocks of code within Yul, defined by curly brackets. They have their own scope and any variable
+declared inside a block cannot be accessed outside it. Statements inside a block can access outside variables, though.
+
+.. code-block:: yul
+
+    {
+        function foo() -> ret {
+            let g := 0
+            { // This is a code block
+                let r := 7
+                ret := mul(g, r)
+            }
+        }
+    }
+
+
+Variable declaration
+____________________
+
+Variables can be declared in Yul using the `let` keyword. Multiple variables can be declared at the same line
+if there is no initializer or the initializer is a functon that returns multiple values.
+
+The default type for variables in Yul is ``u256``. If you want to declare a variable with another type, use the colon.
+Note that if the variable type and the type of the right hand side of the assignment disagree, there will be an implicit
+cast.
+
+.. code-block:: yul
+
+    {
+        let a, b, c
+        let d := funCall()
+        let e : u64 := funcCall()
+        let g, h, i := multipleReturns()
+        let j : u32, k : u8 := manyReturns()
+    }
+
+Assignments
+___________
+
+Variables can be assignment using the ``:=`` operator. If the types do not match,
+the compiler performs an implicit cast, so that the right hand side type matches that of the variable.
+Multiple variables can be assigned in a single line if the right hand side is a function call that returns multiple
+values.
```

Comment:
Does the compiler ever produce an error/warning?

Reply:
Yes, it does

Reply:
I worry that "implicit cast" won't be understood. How about "converted to the right type, if possible".